Assets/MixedRealityToolkit.Providers/Experimental/WindowsMixedReality/WindowsMixedRealitySpatialAwarenessSceneUnderstandingObserver.cs

```diff
@@ -0,0 +1,1245 @@
+                    {
+                        InstantiateSceneObject(saso);
+                    }
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public override void Destroy()
+        {
+            //Debug.Log("Destroy()");
+            killTokenSource.Cancel();
+            CleanupObserver();
+        }
+
+        #endregion IMixedRealityToolkit
+
+        /// <inheritdoc />
+        protected override void Dispose(bool disposing)
+        {
+            if (disposed)
+            {
+                return;
+            }
+
+            base.Dispose(disposing);
+
+            Suspend();
+
+            if (disposing)
+            {
+                CleanupDebugGameObjects();
+            }
+
+            disposed = true;
+        }
+
+        #region IMixedRealitySpatialAwarenessObserver
+
+        private GameObject observedObjectParent = null;
+
+        protected virtual GameObject ObservedObjectParent => observedObjectParent ?? (observedObjectParent = SpatialAwarenessSystem?.CreateSpatialAwarenessObservationParent("WindowsMixedRealitySceneUnderstandingObserver"));
+
+        /// <inheritdoc/>
+        public override void Resume()
+        {
+            updateTimer.Enabled = true;
+        }
+
+        /// <inheritdoc/>
+        public override void Suspend()
+        {
+            if (updateTimer != null)
+            {
+                updateTimer.Enabled = false;
+            }
+        }
+
+        #endregion IMixedRealitySpatialAwarenessObserver
+
+        #region IMixedRealityOnDemandObserver
+
+        public float FirstUpdateDelay { get; set; }
+
+        /// <inheritdoc/>
+        public override void UpdateOnDemand()
+        {
+            if (!MixedRealityToolkit.Instance.ActiveProfile.IsSpatialAwarenessSystemEnabled || (SpatialAwarenessSystem == null))
+            {
+                return;
+            }
+
+            observerState = ObserverState.GetScene;
+        }
+
+        #endregion IMixedRealityOnDemandObserver
+
+        #region Public Profile
+        public Material DefaultMaterial { get; set; } // Need references so they are included for runtime
+        public Material DefaultWorldMeshMaterial { get; set; } // Need references so they are included for runtime
+
+        /// <summary>
+        /// File path to load previously saved scene data
+        /// </summary>
+        public TextAsset SerializedScene
+        {
+            get { return serializedScene; }
+            set
+            {
+                if (serializedScene != value)
+                {
+                    serializedScene = value;
+                    sceneBytes = serializedScene.bytes;
+                }
+            }
+        }
+
+        public bool UpdateOnLoad
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+
+            set
+            {
+                throw new NotImplementedException();
+            }
+        }
+
```

Comment:
if this isn't used in base implementation, then why is it added to the interface? MRTK has gotten bitten a few times by declaring items in interfaces/public definitions and then not needing it/having to clean it up. 
nit: if keeping, add inherit doc as well as simplify. I believe this should work
public bool UpdateOnLoad
{
get => throw new NotImplementedException();
set => throw new NotImplementedException();
}